### ISSUE TITLE
Script syntax error

### ISSUE THREAD

**Issue by basdelfos** [fedorcomander/kissfc-tx-lua-scripts#11]
I get an error `Script syntax error` when starting the script on a Taranis with latest OpenTX 2.2.0RC~~15~~16 build. I downloaded the lua scripts from master and copied the `common` folder and `KissFC.lua` to the TELEMETRY folder on the TARANIS SD. 

I'm out for a couple of days, if you need some debug information, please let me know.

**Comment by basdelfos** [comment]
I have RC16 (N369)

**Comment by fedorcomander** [comment]
i cant just copy common. u need to build scripts....

**Comment by fedorcomander** [comment]
Try minified scripts from here https://github.com/fedorcomander/kissfc-tx-lua-scripts/releases/tag/0.2-mini

**Comment by basdelfos** [comment]
Ah ok, that is different from the Betaflight Lua scripts then.

**Comment by fedorcomander** [comment]
yap. i do compile all to one script. easier for end users.

**Comment by basdelfos** [comment]
I installed the minified script and get a different error now:

`Script panic`
`Not enough memory`

I don't have an connection to a receiver at the moment, could that be problem?

**Comment by fedorcomander** [comment]
do u have any other scripts running???

**Comment by basdelfos** [comment]
Yes, the Kiss Battery Lua script. I will try to disable that and see if it works

**Comment by basdelfos** [comment]
Hmm, still the same error. Probably something broke on with RC16 OpenTx release.
I somehow need to find the RC14 release, Oscar used this version while making the blog post on his site. Companion only lets you download the latest nightly build, that is RC16.

**Comment by fedorcomander** [comment]
Hm... try to use luac (download RC with luac checkbox enabled), then load lua script to opentx simulator. put result KissFC.luac file on the radio.

**Comment by basdelfos** [comment]
Tried today the newest RC17 N370 that came out on 27-04-2017, but still same result.
Then tried your suggestion with the precompiled luac script. It works successfully in the simulator, but not on the Taranis.

On the Taranis I get the following error (only installed the .luac and removed the .lua):
`Script syntax error`
`TELEMETRY/KissFC.luac`
`incompatible precompiled chunk`

**Comment by awolf78** [comment]
Did you use RC17 companion?

**Comment by basdelfos** [comment]
Yes, I'm using RC17 Companion

**Comment by awolf78** [comment]
Well, if you are 100% you checked "luac" when you flashed your Taranis - it must be an issue with OpenTX.
As a workaround you could try: 

- Deactivate all other scripts
- Upload lua script and keep incompatible luac
- Hopefully it recompiles it on the radio
- Delete lua file (but keep luac)
- Do the same with the others

I am using RC14 on my Q7 and have my modded battery script and KISS FC script (with TPA) running at the same time. It is possible, but a bit of a hassle to get there (need to use only luac files).

**Comment by basdelfos** [comment]
I fixed it. 

First I tried with a clean model created with OpenTX RC17 and there the lua script seems to work for some reason. Then I manually copied over all the settings from the old model to the clean model and it still works fine. 

So, for some reason an old model, migrated from previous versions did not work together with the lua script (note: worked with the battery lua script).

I don't know if this is an isolated issue, but when more people are reporting this issue, it seems that the solution is to create a clean model on the Taranis and copy over the settings.
